### PR TITLE
HW #5 (Application architecture)

### PR DESCRIPTION
**Pull request by eluyema**
- Diagram of application architecture
- Test architecture

### CODE REVIEW

**Comment by cholombytko** [comment]
Your architecture diagram looks rightly and well to me, gj!

**Comment by eluyema** [comment]
image

**Comment by cholombytko** [comment]
You implemented nice architecture testing workflow, gj!

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-**This project use DDD architecture, node.js, nest.js, prisma, nodemailer.**
+**This project use Clean architecture, node.js, nest.js, prisma, nodemailer.**
```
Comment (Larisa-Antsifrova) [inline README.md:1]: That's a nice skill to update documentation whenever system has some changes! 🔥

**Comment by Larisa-Antsifrova** [comment]
Thanks for the architecture diagram. It is quite comprehensive 😎 I would suggest adding some arrows to show dependency relationships between the entities and different layers.

File: arch-test/test.arch-spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import 'tsarch/dist/jest';
+
+import { filesOfProject } from 'tsarch';
+
+const skipTestFilesPattern = '^(?!.*\\.(spec|int-spec)\\.ts$).*$';
+
+describe('architecture', () => {
+  jest.setTimeout(60000);
```
Comment (Larisa-Antsifrova) [inline arch-test/test.arch-spec.ts]: Just curious: what is the timeout for? and why this number precisely? :)

**Comment by Larisa-Antsifrova** [comment]
> image

Smart move to attach the test result 😉

**Comment by Larisa-Antsifrova** [comment]
And looks like it would be nice to fix a couple of issues found by the linter check 🙃

**Comment by eluyema** [comment]
> Thanks for the architecture diagram. It is quite comprehensive 😎 I would suggest adding some arrows to show dependency relationships between the entities and different layers.

File: arch-test/test.arch-spec.ts (hunk 2 above)
Comment (eluyema) [reply arch-test/test.arch-spec.ts]: It's limit time for this test, it think it's better to add contants to make it more readable

**Pull request merged**